packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -581,4 +594,74 @@ class EmbraceManager: NSObject {
 
         resolve(true)
     }
+    
+    @objc
+    func logNetworkRequest(
+        _ url: String,
+        httpMethod: String,
+        startInMillis: Double,
+        endInMillis: Double,
+        bytesSent: Double,
+        bytesReceived: Double,
+        statusCode: Double,
+        error: String,
+        resolver resolve: RCTPromiseResolveBlock,
+        rejecter reject: RCTPromiseRejectBlock
+    ) {
+        if Embrace.client == nil {
+            reject("RECORD_LOG_NETWORK_REQUEST_ERROR", "Error recording span, Embrace SDK may not be initialized", nil)
+            return
+        }
+
+        let attributes = attributeStringsFrom(dict: [
+            "http.response.status_code": String(Int(statusCode)),
+            "http.request.body.size": String(Int(bytesSent)),
+            "http.response.body.size": String(Int(bytesReceived)),
+            "http.request.method": httpMethod.uppercased(),
+            "url.full": url,
+        ]);
+
+        Embrace.client?.recordCompletedSpan(name: createNetworkSpanName(url: url, httpMethod: httpMethod),
+                                            type: SpanType.networkRequest, parent: nil,
+                                            startTime: dateFrom(ms: startInMillis), endTime: dateFrom(ms: endInMillis),
+                                            attributes: attributes,
+                                            events: eventsFrom(array: []),
+                                            errorCode: errorCodeFrom(str: error))
```

Comment:
similar here since this is user supplied it probably won't match one of the enum values, probably should set it to `.failure` when error is non-empty or whatever the behaviour is on Android

Reply:
done